modules/account-lib/src/coin/sol/ataInitializationBuilder.ts

```diff
@@ -29,7 +31,8 @@ export class AtaInitializationBuilder extends TransactionBuilder {
       if (instruction.type === InstructionBuilderTypes.CreateAssociatedTokenAccount) {
         const ataInitInstruction: AtaInit = instruction;
 
-        this.mint(ataInitInstruction.params.mintAddress);
+        this._mint = ataInitInstruction.params.mintAddress;
+        this.validateMintOrThrow();
```

Comment:
let's move the validate check to your setter function and change this back to `this.mint(ataInitInstruction.params.mintAddress)`.

we should validate mint when we set it.